src/synthetix/StakingRewards.t.sol

```diff
@@ -270,16 +270,213 @@ contract StakingRewardsTest is Test {
         assertEq(rewards.rewardsDuration(), 70 days);
     }
 
-    function testRevertSetRewardsDurationOnActiveDistribution() public {
+    function testIncreaseRewardsDurationOnActiveDistribution() public {
+        uint256 totalStake = 100 * WAD;
+        uint256 totalReward = 123 * WAD;
         setupStakingToken(100 * WAD);
         rewards.stake(100 * WAD);
-
-        setupReward(100 * WAD);
+        setupReward(totalReward);
+        assertEq(rewardGem.balanceOf(address(this)), 0);
+        assertEq(rewardGem.balanceOf(address(rewards)), totalReward);
+        assertEq(gem.balanceOf(address(this)), 0);
+        assertEq(gem.balanceOf(address(rewards)), totalStake);
+        assertEq(rewards.rewardPerTokenStored(), 0);
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), totalReward / 7 days);
+        assertEq(rewards.periodFinish(), block.timestamp + 7 days);
+        assertEq(rewards.rewardsDuration(), 7 days);
 
         skip(1 days);
 
-        vm.expectRevert("Previous rewards period must be complete before changing the duration for the new period");
         rewards.setRewardsDuration(70 days);
+
+        assertEq(rewards.rewardPerTokenStored(), 1 days * (totalReward / 7 days) * WAD / totalStake);
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), 6 days * (totalReward / 7 days) / 70 days);
+        assertEq(rewards.periodFinish(), block.timestamp + 70 days);
+        assertEq(rewards.rewardsDuration(), 70 days);
+
+        skip(70 days);
+
+        rewards.exit();
+
+        uint256 rewardLeft = totalReward % 7 days + 6 days * (totalReward / 7 days) % 70 days; // dust lost due to rewardRate being rounded down
+        uint256 rewardPaid = totalReward - rewardLeft;
+        assertEq(rewardGem.balanceOf(address(this)), rewardPaid);
+        assertEq(rewardGem.balanceOf(address(rewards)), rewardLeft);
+        assertEq(gem.balanceOf(address(this)), totalStake);
+        assertEq(gem.balanceOf(address(rewards)), 0);
+    }
+
+    function testDecreaseRewardsDurationOnActiveDistribution() public {
+        uint256 totalStake = 100 * WAD;
+        uint256 totalReward = 123 * WAD;
+        setupStakingToken(100 * WAD);
+        rewards.stake(100 * WAD);
+        setupReward(totalReward);
+        assertEq(rewardGem.balanceOf(address(this)), 0);
+        assertEq(rewardGem.balanceOf(address(rewards)), totalReward);
+        assertEq(gem.balanceOf(address(this)), 0);
+        assertEq(gem.balanceOf(address(rewards)), totalStake);
+        assertEq(rewards.rewardPerTokenStored(), 0);
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), totalReward / 7 days);
+        assertEq(rewards.periodFinish(), block.timestamp + 7 days);
+        assertEq(rewards.rewardsDuration(), 7 days);
```

Comment:
This might be better to be extracted to an internal function so we can reuse it.

Reply:
Agreed, done.